types/react-navigation/index.d.ts

```diff
@@ -268,6 +268,7 @@ export interface NavigationInitAction {
 export interface NavigationResetAction {
   type?: 'Navigation/RESET',
   index: number,
+  key?: string | null,
```

Comment:
it would be better like this.
`key?: string | null,`   =>  `key?: string,`

The optional value need not to add null type again.